celery/canvas.py

```diff
@@ -1174,8 +1174,9 @@ class chord(Signature):
 
     @classmethod
     def from_dict(cls, d, app=None):
-        args, d['kwargs'] = cls._unpack_args(**d['kwargs'])
-        return _upgrade(d, cls(*args, app=app, **d))
```

Comment:
There are multiple `from_dict` methods in this module. For example [here](https://github.com/celery/celery/pull/4278/files#diff-e27381cb5c64031dd28aa6855c78cfddR862). Aren't these affected by this issue as well?

Reply:
No, in other places it just uses dict values, and only here it modifies it.